adapter/repository/in-memory-simple/src/test/java/org/mobilitydata/gtfsvalidator/db/NoticeMemoryConsumptionTest.java

```diff
@@ -0,0 +1,140 @@
+package org.mobilitydata.gtfsvalidator.db;
+
+import org.junit.jupiter.api.Test;
+import org.mobilitydata.gtfsvalidator.usecase.notice.IntegerFieldValueOutOfRangeNotice;
+import org.mobilitydata.gtfsvalidator.usecase.notice.base.ErrorNotice;
+import org.mobilitydata.gtfsvalidator.usecase.notice.base.InfoNotice;
+import org.mobilitydata.gtfsvalidator.usecase.notice.base.Notice;
+import org.mobilitydata.gtfsvalidator.usecase.notice.base.WarningNotice;
+import org.mobilitydata.gtfsvalidator.usecase.port.ValidationResultRepository;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class NoticeMemoryConsumptionTest {
+
+    private void generateNotices(MockValidationResultRepo mockValidationResultRepo, int numberOfNotices) {
+        for (int i = 0; i < numberOfNotices; i++) {
+            mockValidationResultRepo.addNotice(new IntegerFieldValueOutOfRangeNotice("filename",
+                    "fieldname",
+                    "entity_id",
+                    0,
+                    100,
+                    101)
+            );
+        }
+    }
+
+    @Test
+    public void creationOf100NoticeShouldNotRaiseException() {
+
+        MockValidationResultRepo mockValidationResultRepo = new MockValidationResultRepo();
+
+        generateNotices(mockValidationResultRepo, 100);
+        mockValidationResultRepo.getAll();
+    }
+
+    @Test
+    public void creationOf1000NoticeShouldNotRaiseException() {
+
+        MockValidationResultRepo mockValidationResultRepo = new MockValidationResultRepo();
+
+        generateNotices(mockValidationResultRepo, 1_000);
+        mockValidationResultRepo.getAll();
+
+    }
+
+    @Test
+    public void creationOf10000NoticeShouldNotRaiseException() {
+
+        MockValidationResultRepo mockValidationResultRepo = new MockValidationResultRepo();
+
+        generateNotices(mockValidationResultRepo, 10_000);
+        mockValidationResultRepo.getAll();
+
+    }
+
+    @Test
+    public void creationOf100000NoticeShouldNotRaiseException() {
+
+        MockValidationResultRepo mockValidationResultRepo = new MockValidationResultRepo();
+
+        generateNotices(mockValidationResultRepo, 100_000);
+        mockValidationResultRepo.getAll();
+
+    }
+
+    @Test
+    public void creationOf1_000_000NoticeShouldNotRaiseException() {
+
+        MockValidationResultRepo mockValidationResultRepo = new MockValidationResultRepo();
+
+        generateNotices(mockValidationResultRepo, 1_000_000);
+        mockValidationResultRepo.getAll();
+    }
+
+    @Test
+    public void creationOf3_000_000NoticeShouldRaiseException() {
```

Comment:
If you're trying to make sure that this method throws an exception, then it should be annotated like:

`@Test(expected = OutOfMemoryError.class)`

...instead of catching the exception. So, there shouldn't be a try/catch block.

Reply:
Oh, thks for that!

Reply:
It seems though things changed a little for JUnit5
https://howtodoinjava.com/junit5/expected-exception-example/

Reply:
@fabrice-v Oh, good point! Looks like you could use either (i.e., annotation on the method should still work), but `assertThrows()` allows more granularity.